src/constants.js

```diff
@@ -59,7 +59,11 @@ function getPreferredCacheDirectories(): Array<string> {
     preferredCacheDirectories.push(getDirectory('cache'));
   }
 
-  preferredCacheDirectories.push(path.join(os.tmpdir(), '.yarn-cache'));
+  if (process.getuid) {
+    preferredCacheDirectories.push(path.join(os.tmpdir(), `.yarn-cache-${process.getuid()}`));
+  }
```

Comment:
At this point, it may be a better idea to make this an `if`/`else` block so we always use the user-specific temp fallback. I'm fine with this version too tho, so approving :)